### PR TITLE
Blog App: Validations and model specs

### PR DESCRIPTION
**Pull request by arnoldnekemiah**
1.  Added the following validations:

        For the User model:
            Name must not be blank.
            PostsCounter must be an integer greater than or equal to zero.
        For the Post model:
            Title must not be blank.
            Title must not exceed 250 characters.
            CommentsCounter must be an integer greater than or equal to zero.
            LikesCounter must be an integer greater than or equal to zero.
   2.  Added unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
```
Comment (FrederickMih) [inline spec/models/post_spec.rb:3]: - [ ] Hi @arnoldnekemiah, it seems we didn't test the scenario where a post shouldn't be created if the title exceed 250 characters :+1: I think this is very important and demands a test for it as you can clearly see in the requirements.
Screenshot from 2023-10-27 13-58-41

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  it 'is not valid without a title' do
+    post = Post.new(author: user, comments_counter: 0, likes_counter: 0)
+    expect(post).to_not be_valid
+  end
+
+  it 'is not valid if comments_counter is negative' do
+    post = Post.new(title: 'A Title', author: user, comments_counter: -1, likes_counter: 0)
+    expect(post).to_not be_valid
```
Comment (FrederickMih) [inline spec/models/post_spec.rb:18]: - [ ] Some of your tests are not running as you can see on the screenshot below:
Screenshot from 2023-10-27 14-05-38

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  it 'is not valid if likes_counter is not an integer' do
+    post = Post.new(title: 'A Title', author: user, comments_counter: 0, likes_counter: 2.5)
+    expect(post).to_not be_valid
+  end
+
+  it 'returns the 5 most recent comments' do
+    post = create(:post) # Use FactoryBot to create a post with comments
```
Comment (FrederickMih) [inline spec/models/post_spec.rb]: - [ ] Let's kindly create a post and at least 6 comments for that post and show that this method(most_recent_comments) returns the 5 most recent comments for that post. We just verify if your tests are well constructed and running without errors. Kindly do the same with all the public methods.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @arnoldnekemiah, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- Validation tests for User model runs with success :heavy_check_mark: 


## Required Changes ♻️

- [ ] Hi @arnoldnekemiah, let's kindly make sure that everyone can contribute to our project with ease by fixing the issue link. Right now the link to the issues page shows 404 error page.

_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @arnoldnekemiah 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- Test updated and are passing   ✅ 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**